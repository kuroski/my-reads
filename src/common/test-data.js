const testBooks = {
  books: [
    {
      title: 'Book Title 1',
      subtitle: 'Book Subtitle 1',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      authors: ['Author One', 'Author Two'],
      imageLinks: { thumbnail: 'http://img01.jpg' },
      id: 'PGR2AwAAQBAJ',
      shelf: 'wantToRead'
    },
    {
      title: 'Book Title 2',
      subtitle: 'Book Subtitle 2',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit2.',
      authors: ['Mist Misterious'],
      imageLinks: { thumbnail: 'http://img02.jpg' },
      id: 'PGR2AwAAQBAJ2',
      shelf: 'read'
    },
    {
      title: 'Book Title 3',
      subtitle: 'Book Subtitle 3',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit3.',
      authors: ['Lucky Guy', 'Telonious Monk', 'Rapid Fast'],
      imageLinks: { thumbnail: 'http://img03.jpg' },
      id: 'PGR2AwAAQBAJ3',
      shelf: 'wantToRead'
    }
  ]
};

export { testBooks };
