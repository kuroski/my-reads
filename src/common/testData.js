import { shelfState } from './shelfState';

const testBooks = {
  books: [
    {
      title: 'Book Title 1',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      authors: ['Author One', 'Author Two'],
      imageLinks: { thumbnail: 'http://img01.jpg' },
      id: 'PGR2AwAAQBAJ',
      shelf: shelfState.WANT_TO_READ
    },
    {
      title: 'Book Title 2',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit2.',
      authors: ['Mist Misterious'],
      imageLinks: { thumbnail: 'http://img02.jpg' },
      id: 'PGR2AwAAQBAJ2',
      shelf: shelfState.READ
    },
    {
      title: 'Book Title 3',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit3.',
      authors: ['Lucky Guy', 'Telonious Monk', 'Rapid Fast'],
      imageLinks: { thumbnail: 'http://img03.jpg' },
      id: 'PGR2AwAAQBAJ3',
      shelf: shelfState.WANT_TO_READ
    }
  ]
};

export { testBooks };
