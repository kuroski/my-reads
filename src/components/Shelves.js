import React from 'react';
import PropTypes from 'prop-types';
import BookShelf from './BookShelf';

const Shelves = props => {
  const renderBookSHelves = () => {
    if (!props.shelves || props.shelves.length === 0)
      return <div className="empty-message">Oops, no shelve created</div>;

    return props.shelves.map(shelve => (
      <BookShelf
        key={shelve.name}
        name={shelve.name}
        shelveId={shelve.value}
        shelves={props.shelves}
        books={props.books}
        onMove={props.onMove}
      />
    ));
  };

  return <div>{renderBookSHelves()}</div>;
};

Shelves.propTypes = {
  shelves: PropTypes.array.isRequired,
  books: PropTypes.array.isRequired,
  onMove: PropTypes.func.isRequired
};

export default Shelves;
