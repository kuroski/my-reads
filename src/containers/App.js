import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import SearchPage from './SearchPage';
import Shelves from './Shelves';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <Route path="/search" render={() => <SearchPage />} />
        <Route
          exact
          path="/"
          render={() => (
            <div className="list-books">
              <div className="list-books-title">
                <h1>MyReads</h1>
              </div>
              <div className="list-books-content">
                <Shelves />
              </div>

              <div className="open-search">
                <Link to="/search">Add a book</Link>
              </div>
            </div>
          )}
        />
      </div>
    );
  }
}

export default App;
