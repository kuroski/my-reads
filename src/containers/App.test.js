import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import App from './App';
import Shelves from '../components/Shelves';
import { shelfState } from '../common/shelfState';
import { testBooks } from '../common/testData';

describe('Shelves Container', () => {
  const build = () => {
    return mount(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
  };

  it('renders App Container correctly', () => {
    const ShelvesTree = renderer
      .create(
        <MemoryRouter>
          <App />
        </MemoryRouter>
      )
      .toJSON();
    expect(ShelvesTree).toMatchSnapshot();
  });

  it('must call "onMove" prop when shelf is changed', () => {
    const wrapper = build();
    wrapper.setState({ books: testBooks.books });
    const shelves = wrapper.find(Shelves);

    expect(wrapper.state('books')[0].shelf).toEqual(shelfState.WANT_TO_READ);
    shelves.prop('onMove')(testBooks.books[0].id, shelfState.READ);
    expect(wrapper.state('books')[0].shelf).toEqual(shelfState.READ);
  });
});
