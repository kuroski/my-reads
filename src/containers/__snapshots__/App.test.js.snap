// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shelves Container renders App Container correctly 1`] = `
<div
  className="app"
>
  <div
    className="list-books"
  >
    <div
      className="list-books-title"
    >
      <h1>
        MyReads
      </h1>
    </div>
    <div
      className="list-books-content"
    >
      <div>
        <div
          className="bookshelf"
        >
          <h2
            className="bookshelf-title"
          >
            Currently Reading
          </h2>
          <div
            className="bookshelf-books"
          >
            <ol
              className="books-grid"
            >
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 1"
                      className="book-cover"
                      height="193"
                      src="http://img01.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="currentlyReading"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 1
                  </div>
                  <div
                    className="book-authors"
                  >
                    Author One, Author Two
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 2"
                      className="book-cover"
                      height="193"
                      src="http://img02.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="currentlyReading"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 2
                  </div>
                  <div
                    className="book-authors"
                  >
                    Mist Misterious
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 3"
                      className="book-cover"
                      height="193"
                      src="http://img03.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="currentlyReading"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 3
                  </div>
                  <div
                    className="book-authors"
                  >
                    Lucky Guy, Telonious Monk, Rapid Fast
                  </div>
                </div>
              </li>
            </ol>
          </div>
        </div>
        <div
          className="bookshelf"
        >
          <h2
            className="bookshelf-title"
          >
            Want to Read
          </h2>
          <div
            className="bookshelf-books"
          >
            <ol
              className="books-grid"
            >
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 1"
                      className="book-cover"
                      height="193"
                      src="http://img01.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="wantToRead"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 1
                  </div>
                  <div
                    className="book-authors"
                  >
                    Author One, Author Two
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 2"
                      className="book-cover"
                      height="193"
                      src="http://img02.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="wantToRead"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 2
                  </div>
                  <div
                    className="book-authors"
                  >
                    Mist Misterious
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 3"
                      className="book-cover"
                      height="193"
                      src="http://img03.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="wantToRead"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 3
                  </div>
                  <div
                    className="book-authors"
                  >
                    Lucky Guy, Telonious Monk, Rapid Fast
                  </div>
                </div>
              </li>
            </ol>
          </div>
        </div>
        <div
          className="bookshelf"
        >
          <h2
            className="bookshelf-title"
          >
            Read
          </h2>
          <div
            className="bookshelf-books"
          >
            <ol
              className="books-grid"
            >
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 1"
                      className="book-cover"
                      height="193"
                      src="http://img01.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="read"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 1
                  </div>
                  <div
                    className="book-authors"
                  >
                    Author One, Author Two
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 2"
                      className="book-cover"
                      height="193"
                      src="http://img02.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="read"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 2
                  </div>
                  <div
                    className="book-authors"
                  >
                    Mist Misterious
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="book"
                >
                  <div
                    className="book-top"
                  >
                    <img
                      alt="Book Title 3"
                      className="book-cover"
                      height="193"
                      src="http://img03.jpg"
                      width="128"
                    />
                    <div
                      className="book-shelf-changer"
                    >
                      <select
                        onChange={[Function]}
                        value="read"
                      >
                        <option
                          disabled={true}
                          value="none"
                        >
                          Move to...
                        </option>
                        <option
                          className="book-shelf-option"
                          value="currentlyReading"
                        >
                          Currently Reading
                        </option>
                        <option
                          className="book-shelf-option"
                          value="wantToRead"
                        >
                          Want to Read
                        </option>
                        <option
                          className="book-shelf-option"
                          value="read"
                        >
                          Read
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    className="book-title"
                  >
                    Book Title 3
                  </div>
                  <div
                    className="book-authors"
                  >
                    Lucky Guy, Telonious Monk, Rapid Fast
                  </div>
                </div>
              </li>
            </ol>
          </div>
        </div>
      </div>
    </div>
    <div
      className="open-search"
    >
      <a
        href="/search"
        onClick={[Function]}
      >
        Add a book
      </a>
    </div>
  </div>
</div>
`;
